### PR TITLE
adjust proficiency ratios

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
Bugfixes "Standardizes time_factor and fail_factor for proficiency books"

#### Purpose of change
I was unclear in my initial specifications for how I expected time_factor and fail_factor to work in proficiency book mitigation. This led to two different ways of doing it, so some of our books are dramatically too good at mitigating proficiencies

#### Describe the solution
Audits proficiency books to all have the same standard effect, where a higher time_factor or fail_factor is better.

#### Describe alternatives you've considered
n/a

#### Testing
tk

#### Additional context
I will try to update documentation in this PR if I have time, along with a standard set of suggested values, something like:

time_factor:
* 0.1 - an obtuse reference book that is hard to follow
* 0.2 - a good, well laid-out reference. Standard.
* 0.3 - an excellent, easy to read book that gives lots of efficient tips
* 0.4 - a book designed to expedite your crafting process and teach you more about speedy work than anything else.

fail_factor:
* 0.1 - gives some vague tips to help you learn the proficiency, along the level of a periodical or blog post
* 0.2 to 0.3 - gives good, useful tips, like you'd find in a normal reference book
* 0.4 - gives excellent, high quality advice for beginners that could help them from the ground up
* 0.5 - top level reference book with not just advice for what to do, but good advice on how to trouble shoot it

In general, a book that has too many recipes should be assumed to have lower values here as we can assume it brevifies and simplifies the process. The books with the best mitigation should be non-periodical full books with a few recipes that are well documented. These stats should usually be handled as a tradeoff with a good book helping either both a little, or one more than the other: a book that includes a ton of detail on troubleshooting is presumed to be harder to reference and scan through, while a book that is designed for quick reference and speed tips doesn't have as much detail to avoid traps. This is a guideline though, not a rule.

### CODE REVIEW

**Comment by actual-nh** [comment]
So, for a book-length work with relatively few recipes, time_factor+fail_factor = 0.7 would be reasonable for most? If in doubt, time_factor would be correlated with fun, I would think.

**Comment by I-am-Erk** [comment]
> So, for a book-length work with relatively few recipes, time_factor+fail_factor = 0.6-0.7 would be reasonable for most? If in doubt, time_factor would be correlated with fun, I would think.
> 
> EDIT: Other factors could be how much it teaches you skillwise (minimum vs maximum skill levels), how much INT is required, and how many proficiencies.

I don't know if I want to get that precise but it would be worth mentioning a guideline that for periodicals time plus fail should probably not be >0.3, for average references not >0.5, and 0.7 or so should be considered pretty much maximum

File: data/json/items/book/fabrication.json [hunk 1]
```diff
@@ -136,7 +136,7 @@
     "skill": "fabrication",
     "required_level": 3,
     "max_level": 5,
-    "proficiencies": [ { "proficiency": "prof_bowyery", "time_factor": 0.75, "fail_factor": 0.5 } ],
+    "proficiencies": [ { "proficiency": "prof_bowyery", "time_factor": 0.15, "fail_factor": 0.5 } ],
```
Comment (actual-nh) [inline data/json/items/book/fabrication.json:139]: Either this one shouldn't be quite this good (I'd go with fail_factor 0.45), or the Fletcher's Friend should be a bit better (I'd go with fail_factor 0.4), considering that they're only 1 level apart.

File: data/json/items/book/fabrication.json [hunk 2]
```diff
@@ -328,7 +328,7 @@
     "skill": "fabrication",
     "required_level": 5,
     "max_level": 10,
-    "proficiencies": [ { "proficiency": "prof_bladesmith", "time_factor": 0.65, "fail_factor": 0.6, "include_prereqs": false } ],
+    "proficiencies": [ { "proficiency": "prof_bladesmith", "time_factor": 0.2, "fail_factor": 0.3, "include_prereqs": false } ],
```
Comment (actual-nh) [inline data/json/items/book/fabrication.json:331]:
```suggestion
    "proficiencies": [ { "proficiency": "prof_bladesmith", "time_factor": 0.25, "fail_factor": 0.3, "include_prereqs": false } ],
```
"Step-by-step photo instructions", plus consideration of other stats.

File: data/json/items/book/fabrication.json [hunk 3]
```diff
@@ -348,7 +348,7 @@
     "skill": "fabrication",
     "required_level": 5,
     "max_level": 10,
-    "proficiencies": [ { "proficiency": "prof_bladesmith", "time_factor": 0.75, "fail_factor": 0.75, "include_prereqs": true } ],
+    "proficiencies": [ { "proficiency": "prof_bladesmith", "time_factor": 0.2, "fail_factor": 0.2, "include_prereqs": true } ],
```
Comment (actual-nh) [inline data/json/items/book/fabrication.json:351]:
```suggestion
    "proficiencies": [ { "proficiency": "prof_bladesmith", "time_factor": 0.25, "fail_factor": 0.3, "include_prereqs": true } ],
```

**Comment by kevingranade** [comment]
LGTM, is there anything keeping it in draft?

File: data/json/items/book/fabrication.json [hunk 4]
```diff
@@ -369,7 +369,7 @@
     "skill": "fabrication",
     "required_level": 4,
     "max_level": 7,
-    "proficiencies": [ { "proficiency": "prof_welding", "time_factor": 0.75, "fail_factor": 0.6, "include_prereqs": true } ],
+    "proficiencies": [ { "proficiency": "prof_welding", "time_factor": 0.25, "fail_factor": 0.3, "include_prereqs": true } ],
     "intelligence": 9,
     "time": "30 m"
```
Comment (actual-nh) [inline data/json/items/book/fabrication.json:374]:
```suggestion
    "proficiencies": [ { "proficiency": "prof_welding", "time_factor": 0.2, "fail_factor": 0.35, "include_prereqs": true } ],
    "intelligence": 9,
    "time": "45 m",
    "fun": -1
```
"rather technical"

File: data/json/items/book/fabrication.json [hunk 5]
```diff
@@ -390,8 +390,8 @@
     "max_level": 2,
     "intelligence": 5,
     "proficiencies": [
-      { "proficiency": "prof_carpentry_basic", "time_factor": 0.4, "fail_factor": 0.8, "include_prereqs": true },
-      { "proficiency": "prof_plumbing", "time_factor": 0.5, "fail_factor": 0.8, "include_prereqs": true }
+      { "proficiency": "prof_carpentry_basic", "time_factor": 0.2, "fail_factor": 0.2, "include_prereqs": true },
+      { "proficiency": "prof_plumbing", "time_factor": 0.2, "fail_factor": 0.3, "include_prereqs": true }
```
Comment (actual-nh) [inline data/json/items/book/fabrication.json:394]:
```suggestion
      { "proficiency": "prof_carpentry_basic", "time_factor": 0.2, "fail_factor": 0.3, "include_prereqs": true },
      { "proficiency": "prof_plumbing", "time_factor": 0.2, "fail_factor": 0.2, "include_prereqs": true }
```
Given that it's described as for the "novice carpenter".

File: data/json/items/book/fabrication.json [hunk 6]
```diff
@@ -456,8 +456,8 @@
     "skill": "fabrication",
     "max_level": 3,
     "proficiencies": [
-      { "proficiency": "prof_carpentry_basic", "time_factor": 0.5, "fail_factor": 0.4, "include_prereqs": true },
-      { "proficiency": "prof_plumbing", "time_factor": 0.5, "fail_factor": 0.6, "include_prereqs": true }
+      { "proficiency": "prof_carpentry_basic", "time_factor": 0.1, "fail_factor": 0.4, "include_prereqs": true },
+      { "proficiency": "prof_plumbing", "time_factor": 0.1, "fail_factor": 0.2, "include_prereqs": true }
```
Comment (actual-nh) [inline data/json/items/book/fabrication.json:460]:
```suggestion
      { "proficiency": "prof_plumbing", "time_factor": 0.1, "fail_factor": 0.3, "include_prereqs": true }
```
More similar to values for 101_carpentry.

File: data/json/items/book/tailor.json [hunk 7]
```diff
@@ -144,6 +147,10 @@
     "looks_like": "textbook_firstaid",
     "color": "blue",
     "skill": "tailor",
+    "proficiencies": [
+      { "proficiency": "prof_closures", "time_factor": 0.1, "fail_factor": 0.25 },
+      { "proficiency": "prof_elastics", "time_factor": 0.1, "fail_factor": 0.25 }
```
Comment (actual-nh) [inline data/json/items/book/tailor.json:152]:
```suggestion
      { "proficiency": "prof_closures", "time_factor": 0.1, "fail_factor": 0.3 },
      { "proficiency": "prof_elastics", "time_factor": 0.1, "fail_factor": 0.3 }
```
But also increase the time to 55 minutes. The fail_factor might be increased further depending on how many recipes it has.

**Review by actual-nh — COMMENTED** [review]
These are not taking into account how many recipes the book has, so may need adjusting for that.

File: data/json/items/book/tailor.json [hunk 8]
```diff
@@ -63,8 +66,8 @@
     "skill": "tailor",
     "max_level": 3,
     "proficiencies": [
-      { "proficiency": "prof_closures", "time_factor": 0.5, "fail_factor": 0.75 },
-      { "proficiency": "prof_elastics", "time_factor": 0.5, "fail_factor": 0.75 }
+      { "proficiency": "prof_closures", "time_factor": 0.2, "fail_factor": 0.25 },
+      { "proficiency": "prof_elastics", "time_factor": 0.1, "fail_factor": 0.25 }
```
Comment (actual-nh) [inline data/json/items/book/tailor.json:70]:
```suggestion
      { "proficiency": "prof_elastics", "time_factor": 0.2, "fail_factor": 0.15 }
```
It only takes 18 minutes to read, intelligence required is 4, etc.

File: data/json/items/book/fabrication.json (hunk 3 above)
Comment (actual-nh) [inline data/json/items/book/fabrication.json:351]:
```suggestion
    "proficiencies": [ { "proficiency": "prof_bladesmith", "time_factor": 0.2, "fail_factor": 0.3, "include_prereqs": true } ],
```
Description + other stats.

Comment (I-am-Erk) [reply data/json/items/book/fabrication.json:351]: Include_prereqs means it will apply its bonuses to lower level proficiencies if you lack the higher level ones. I wouldn't figure it too much into the balance equations although it does make a book a better proficiency reference
